### ISSUE TITLE
[Enchancement] Add a way to set map language tag

### ISSUE THREAD

**Issue by w1zz4** [Jackett/Jackett#11014]
There is already an option to change MULTI for a few Tracker (Like yggtorrents). But there are some custom language tags for regionalized version. 

A good example is VFQ which stand for Version Française Québécois (French Canadian Version). It will be good to have a text box where someone can just put VFQ then choose the corresponding language FRENCH so the client get the right language information.

**Comment by xfouloux** [comment]
If i understand correctly you want the parser to ADD or REPLACE VFQ by FRENCH like in this example :

on tracker : Subito.texto.SAISON.5.VFQ.WEB-DL.480p.x264-ROLLED
on jackett : 
ADD  -> Subito.texto.SAISON.5.VFQ.FRENCH.WEB-DL.480p.x264-ROLLED
REPLACE -> Subito.texto.SAISON.5.FRENCH.WEB-DL.480p.x264-ROLLED

Maybe also for VFF ?

**Comment by w1zz4** [comment]
The issue right now is that VFF or VFQ are not considered as language by download app.  

At first I thought a settings like the one in yggtorrents to map MULTI with a langague could be a good idea. But now that you are talking about adding FRENCH avec VFQ or VFF is a better idea. You still can see if it's quebecois or not release but downloader will see it as french

**Comment by xfouloux** [comment]
Let say we add a checkbox option that say

Add .FRENCH to any instance of word "VFF" / "VFQ" 

-- moreover

i'll probably make a PR soon for this then, and using this to add more regex to all indexer that have this option, once there is approval of modification about the following @garfield69 @ilike2burnthing 

for multi it match only "(?i)(\\smulti\\s)"
if MULTI is at end of title, it won't match so need to add another regex "(?i)(\\smulti$)"

Also need to match this terms

VFF+VO
VFF + VO
VO+VFF
VO + VFF
TRUEFRENCH + VO
TRUEFRENCH+VO

we could include this too but then it would interfere with the request in this issue
VO+VFQ
VO + VFQ

same for vostfr regex it matches 

"(?i)(\\svostfr\\s)" and "(?i)(\\ssubfrench\\s)"
we need to match at end too
"(?i)(\\svostfr$)" and "(?i)(\\ssubfrench$)"

**Comment by w1zz4** [comment]
This is cool and address my request properly.

But wouldn’t be better to add a general option available to all trackers where user can do their own matching regex and replacent string?

**Comment by xfouloux** [comment]
> This is cool and address my request properly.
> 
> But wouldn’t be better to add a general option available to all trackers where user can do their own matching regex and replacent string?

Yes i updated my comments multiple time regarding this

**Comment by ilike2burnthing** [comment]
@xfouloux are you planning to have this as an optional user-configurable setting which can be added per indexer, but without having to also add lines upon lines of `replace` and `re_replace` title filters (because this would be called from elsewhere)?

YGG being the obvious example, but also much more extensive versions such as HD-Forever: https://github.com/Jackett/Jackett/blob/eaf0513605c3f8e982c36617ca43555f2ac7a5e4/src/Jackett.Common/Definitions/hdforever.yml#L199-L254

**Comment by xfouloux** [comment]
@ilike2burnthing  yes i would like this to be an optional user-configurable setting in the settings block

you would have something like 
```
settings:
  - name: username
    type: text
    label: Username
  - name: password
    type: password
    label: Password
  - name: multiandvostfrregex
```

this would then display automatically in the indexer config the checkboxes and select lists and do the regexes after on results, this way we would have only 1 place in the code to change regex and match patterns for this, instead of in each indexer, where as you can see, hdforever has better regex than ygg or sharewood for example.

**Comment by ilike2burnthing** [comment]
The first option would seem more suitable.

Just wondering if this could be made to work with other languages as well (Spanish comes to mind), e.g.
```
  - name: multilang
    arg: french
```
and rules would be used specific to each language.

Obviously `arg:` doesn't work in settings, so just using it to give a rough idea.

**Comment by xfouloux** [comment]
Indeed ! Obviously a great idea !

then we would create a file in the dir Jackett/src/Jackett.Common/Utils/
for each supported lang, and put functions in there i think
In /src/Jackett.Common/Utils/ParseUtil.cs we would have the method, calling function with lang and args

definitely worth it

**Comment by xfouloux** [comment]
reviewing the code, it's easier, and maybe better to add it as a global option, because in settings bloc, you have to give it a name, type, and label and optionnaly a default param, because it's how it's constructed; Moreover, you would be able to add multiple blocs using multiple times the same type so meh.

i'm currently trying to add this "extraoption" yaml selector, it would look like this
```
---
id: sharewood
name: Sharewood
description: "sharewood is a Semi-Private FRENCH Torrent Tracker for GENERAL"
language: fr-fr
...
...
caps:
  categorymappings:
    - {id: "Films", cat: Movies, desc: "Films"}
...
  modes:
    search: [q]
...
...
extraoptions:
  - freeleech (bool)
  - regexlang: string LANG
...
...
settings:
  - name: username
```

the "freeleech" option would obviously add the checkbox freeleech to be checked or not on indexer config, and then filter results to only display downloadfactor=0 results

the "regexlang" option would display thebloc 
![image](https://user-images.githubusercontent.com/9778409/111593958-9bd8b880-881e-11eb-9a45-e3b35b70bf78.png)
and apply regex depending on the LANG selected as arg.

what other options would be great there that are redundant ?

**Comment by ilike2burnthing** [comment]
Would `freelech` be doable, given that it's usually specific to the engine or tracker, in some cases just filtering results after they're returned? Or would it just replace:
```
  - name: freeleech
    type: checkbox
    label: Search freeleech only
    default: false
```
If the latter, I don't think it's necessary.

**Comment by xfouloux** [comment]
well my idea about freeleech, is yeah it would automatically add a setting bloc
```
  - name: freeleech
    type: checkbox
    label: Search freeleech only
    default: false
```
but also, to know if a file is freeleech or not, you just have to filter on the return of value downloadratio, which may or may not be modified by the filtering or whatever you do regarding the tracker. But if it is not configured, the value is 1 by default. So you just have to filter results and display only results that have downloadratio = 0 when freeleech checkbox is ticked.

So the way to modify the output of the downloadratio is up to the tracker config

**Comment by Tristan-2i** [comment]
hi, did you find a solution ?
is there a way to download only vff and not vfq ?

**Comment by w1zz4** [comment]
> hi, did you find a solution ?
> is there a way to download only vff and not vfq ?

I moved to a alternate software since. This was not the main reason btw.

**Comment by Tristan-2i** [comment]
> > hi, did you find a solution ?
> > is there a way to download only vff and not vfq ?
> 
> I moved to a alternate software since. This was not the main reason btw.

Oh.. I found that the option to map VFQ as FRENCH is available for "lesaloon" indexer so I think it can be the same for ygg.
My problem is that i don't want to download VFQ version.
For exemple a file like MULTI.VFQ is transformed like FRENCH.VFQ but I don't want VFQ files only VFF, what should I do ?

**Comment by ilike2burnthing** [comment]
Create a new release profile with a `must not contain` restriction for `VFQ` in Sonarr.

**Comment by Tristan-2i** [comment]
> Create a new release profile with a `must not contain` restriction for `VFQ` in Sonarr.

I'm sorry but how to create a release profile in Jackett ?

**Comment by ilike2burnthing** [comment]
Use `Custom Formats` in Radarr.

**Comment by Tristan-2i** [comment]
> Use `Custom Formats` in Radarr and enable the `Negate` setting for it.

Thank you I discover the functionalities of the personalized format.
For my specific use I just found it under Settings/Indexers/Restrictions
